aws/resource_aws_msk_cluster.go

```diff
@@ -0,0 +1,308 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kafka"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"log"
+	"time"
+)
+
+func resourceAwsMskCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMskClusterCreate,
+		Read:   resourceAwsMskClusterRead,
+		Delete: resourceAwsMskClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(120 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"client_subnets": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+			},
+			"broker_instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"broker_security_groups": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+			},
+			"broker_volume_size": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"encrypt_rest_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+				ForceNew: true,
+			},
+			"enhanced_monitoring": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  kafka.EnhancedMonitoringDefault,
+				ValidateFunc: validation.StringInSlice([]string{
+					kafka.EnhancedMonitoringDefault,
+					kafka.EnhancedMonitoringPerBroker,
+					kafka.EnhancedMonitoringPerTopicPerBroker,
+				}, true),
+				ForceNew: true,
+			},
+			"kafka_version": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
This should be required and without the default which would force frequent updates to the provider as the available versions are updated by AWS.